manta-crypto/src/eclair/alloc.rs

```diff
@@ -191,7 +355,12 @@ where
     }
 }
 
-impl<COM, T> Allocate<COM> for T where T: ?Sized {}
+impl<COM, T> Allocate<COM> for T
+where
+    COM: ?Sized,
+    T: ?Sized,
+{
+}
```

Comment:
Naive question: doesn't this trait with implementation for any (possibly ?Sized) generic T allow us to implement both Constant and Variable for any T in an analogous way to what was done above, i.e., by using as_constant, as_known and as_unknown?

Reply:
Because we give this implementation, this trait is in-fact _not_ implementable by anyone else (since we've given an implementation $\forall\text{ }\textsf{T}$). So for future implementers of `Constant` and `Variable` they will automatically inherit an `Allocate` implementation for their underlying `Constant::Type` or `Variable::Type` which comes with an `as_constant` and `as_known` methods. No one will be able to give a different implementation of these methods, they are just helper functions that are useful for reducing boilerplate.